components/playlist/playlist_download_request_manager.cc

```diff
@@ -142,14 +142,31 @@ void PlaylistDownloadRequestManager::GetMedia(content::WebContents* contents) {
   DCHECK(contents && contents->GetPrimaryMainFrame());
 
   const auto& media_detector_script =
-      media_detector_component_manager_->GetMediaDetectorScript();
+      media_detector_component_manager_->GetMediaDetectorScript(
+          contents->GetVisibleURL());
   DCHECK(!media_detector_script.empty());
 
-  contents->GetPrimaryMainFrame()->ExecuteJavaScriptInIsolatedWorld(
+#if BUILDFLAG(IS_ANDROID)
+  content::RenderFrameHost::AllowInjectingJavaScript();
+  contents->GetPrimaryMainFrame()->ExecuteJavaScript(
       base::UTF8ToUTF16(media_detector_script),
       base::BindOnce(&PlaylistDownloadRequestManager::OnGetMedia,
-                     weak_factory_.GetWeakPtr(), contents->GetWeakPtr()),
-      g_playlist_javascript_world_id);
+                     weak_factory_.GetWeakPtr(), contents->GetWeakPtr()));
+#else
+  if (run_script_on_main_world_) {
+    contents->GetPrimaryMainFrame()->ExecuteJavaScriptForTests(
+        base::UTF8ToUTF16(media_detector_script),
+        base::BindOnce(&PlaylistDownloadRequestManager::OnGetMedia,
+                       weak_factory_.GetWeakPtr(), contents->GetWeakPtr()));
+
+  } else {
+    contents->GetPrimaryMainFrame()->ExecuteJavaScriptInIsolatedWorld(
+        base::UTF8ToUTF16(media_detector_script),
+        base::BindOnce(&PlaylistDownloadRequestManager::OnGetMedia,
+                       weak_factory_.GetWeakPtr(), contents->GetWeakPtr()),
+        g_playlist_javascript_world_id);
```

Comment:
it looks like this is set in https://github.com/brave/brave-core/blob/master/browser/playlist/playlist_service_factory.cc#L76 to `ISOLATED_WORLD_ID_CHROME_INTERNAL` which is used for a bunch of chromium injected scripts? should we set our own ID so playlist doesn't conflict with these?

Reply:
Good point! We seem to have ISOLATED_WORLD_ID_BRAVE_INTERNAL so I think this should use that.

Reply:
i started a discussion here - https://bravesoftware.slack.com/archives/C7VLGSR55/p1673046335116019. AFAICT ISOLATED_WORLD_ID_BRAVE_INTERNAL is only used at the moment for cosmetic filters and it seems playlist is roughly the same "trust level" as cosmetic filters, so i think it should be ok. but in the long term we should make it easier to see which worlds are being used for what.